lib/mix/test/mix/dep/lock_test.exs

```diff
@@ -61,4 +61,84 @@ defmodule Mix.Dep.LockTest do
       end
     end
   end
+
+  test "solves single merge conflicts", context do
```

Comment:
We need to have a test checking what happens when there are duplicate keys. For example, :foo 0.1.0 and :foo 0.2.0.